File: libs/perllib/LoxBerry/testing/check_secpin.pl
#!/usr/bin/perl

use LoxBerry::System;

my $entered_pin = $ARGV[1];

print "Checking pin '$entered_pin'...\n";
  
my $result = LoxBerry::System::check_securepin($entered_pin);
  
if ( !$result ) {
	print "SecurePIN OK";
} elsif ( $result == 1 ) {
	print "SecurePIN wrong";
} elsif ( $result == 2 ) {
	print "SecurePIN file could not be opened";
} elsif ( $result == 3 ) {
	print "SecurePIN currently is LOCKED";
} else {
	print "Undefined result $result";
}
print "\n";

File: libs/pythonlib/testing/check_secpin.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""
check_secpin.py - Test check_securepin().
Analogous to perllib/LoxBerry/testing/check_secpin.pl.

Only tests with a deliberately INVALID pin and resets the brute-force counter
file around the call, so no lasting state remains and no real PIN is exposed.
"""

import os
import sys

sys.path.insert(0, os.path.dirname(os.path.dirname(os.path.abspath(__file__))))

from loxberry import system as lb

errfile = "%s/log/system_tmpfs/securepin.errors" % lb.lbhomedir

if os.path.exists(errfile):
    os.unlink(errfile)

result = lb.check_securepin("zzz_invalid_pin_zzz")
print("check_securepin(invalid) -> %r  (None=ok, 1=wrong, 3=locked)" % result)

if os.path.exists(errfile):
    os.unlink(errfile)

print("\nOK")
